server/container_create_linux.go

```diff
@@ -761,3 +771,34 @@ func addShmMount(ctr ctrfactory.Container, sb *sandbox.Sandbox) {
 		Options:     []string{"rw", "bind"},
 	})
 }
+
+// verifyImageSignature checks the signature of the image specified in the mount.
+func (s *Server) verifyImageSignature(ctx context.Context, namespace string, m *types.Mount, status *storage.ImageResult) error {
+	systemCtx, err := s.contextForNamespace(namespace)
+	if err != nil {
+		return fmt.Errorf("get context for namespace: %w", err)
+	}
+
+	if systemCtx.SignaturePolicyPath != "" {
+		// This will likely fail in a container restore case.
+		// This is okay; in part because container restores are an alpha feature,
+		// and it is meaningless to try to verify an image that isn't even an image
+		// (like a checkpointed file is).
+		if m.Image.UserSpecifiedImage == "" {
+			return errors.New("user specified image not specified, cannot verify image signature")
+		}
+
+		var userSpecifiedImageRef references.RegistryImageReference
+
+		userSpecifiedImageRef, err = references.ParseRegistryImageReferenceFromOutOfProcessData(m.Image.UserSpecifiedImage)
+		if err != nil {
+			return fmt.Errorf("unable to get userSpecifiedImageRef from user specified image %q: %w", m.Image.UserSpecifiedImage, err)
+		}
+
+		if err := s.ContainerServer.StorageImageServer().IsRunningImageAllowed(ctx, &systemCtx, userSpecifiedImageRef, status.ID); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
```

Comment:
If that's the code from https://github.com/cri-o/cri-o/blob/e7cdc81463713dd822c2194ea9d2e9b17e31e398/server/container_create.go#L701-L732

then also just call `verifyImageSignature` there.

Reply:
Done